ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -332,10 +332,13 @@ public boolean isOpen() {
     @Override
     public void close()
         throws IOException {
-      isOpen = false;
-      if (!operationCompleted) {
-        abort(new RouterException("The ReadableStreamChannel for blob data has been closed by the user.",
-            RouterErrorCode.ChannelClosed));
+      if (isOpen) {
```

Comment:
Originally, there were no read-modify-write for this variable, so it did not have to be atomic. Now it does.